services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crt/S3NativeClientConfiguration.java

```diff
@@ -110,50 +99,14 @@ public URI endpointOverride() {
         return endpointOverride;
     }
 
-    public Executor futureCompletionExecutor() {
```

Comment:
Removed this `futureCompletionExecutor` block because this is no longer needed. (Should've created a separate PR for this change 😓 I can remove this change if this makes the review difficult)